firefox-ios/Client/Frontend/Home/Wallpaper/Redux/WallpaperState.swift

```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+import Redux
+import Common
+
+struct WallpaperState: ScreenState, Equatable {
+    var windowUUID: WindowUUID
+    var wallpaper: Wallpaper
+
+    init(windowUUID: WindowUUID, wallpaper: Wallpaper = Wallpaper.defaultWallpaper) {
+        self.windowUUID = windowUUID
+        self.wallpaper = wallpaper
+    }
+
+    static let reducer: Reducer<Self> = { state, action in
+        guard action.windowUUID == .unavailable || action.windowUUID == state.windowUUID else {
+            return defaultState(from: state)
+        }
+
+        switch action.actionType {
+        case WallpaperMiddlewareActionType.wallpaperDidInitialize:
+            return handleWallpaperAction(action: action, state: state)
+        case WallpaperMiddlewareActionType.wallpaperDidChange:
+            return handleWallpaperAction(action: action, state: state)
```

Comment:
I think that they could be the same case but they should(?) be different action types since they are different actions? Would it be better if we structured it like:
```
case  WallpaperMiddlewareActionType.wallpaperDidInitialize,
          WallpaperMiddlewareActionType.wallpaperDidChange:
          return handleWallpaperAction(action: action, state: state)
```
So we are not duplicating call sites?

Reply:
ah, sorry I wasn't clear. The code snippet you had above is exactly what I was looking for!

Reply:
I made this change so I will resolve this comment!